src/modm/platform/uart/stm32/uart.hpp.in

```diff
@@ -74,27 +62,19 @@ public:
 		Connector::connect();
 	}
 
+	/// @warning Remember to set word length correctly when using the parity bit!
 	template< class SystemClock, baudrate_t baudrate, percent_t tolerance=pct(1) >
-	static void modm_always_inline
-%% if options["buffered"]
-	initialize(uint32_t interruptPriority = 12, Parity parity = Parity::Disabled)
-%% else
-	initialize(Parity parity = Parity::Disabled)
-%% endif
+	static inline void
+	initialize(Parity parity=Parity::Disabled, WordLength length=WordLength::Bit8)
 	{
-		{{ hal }}::initializeWithBrr(
-				UartBaudrate::getBrr<SystemClock::{{ name }}, baudrate, tolerance>(),
-%% if target["family"] == "f1"
-				parity);
-%% else
-				parity,
-				UartBaudrate::getOversamplingMode(SystemClock::{{ name }}, baudrate));
-%% endif
-%% if options["buffered"]
-		initializeBuffered(interruptPriority);
+		{{ hal }}::initialize<SystemClock, baudrate, tolerance>(parity, length);
```

Comment:
This might fail if `initialize()` is called a second time to change settings because `UE` is still set. With the old version it should have worked. Shouldn't we put a `disableOperation()` before that line to permit it again?

Reply:
Very good point!